autopilot/scanner.go

```diff
@@ -214,36 +214,45 @@ func (s *scanner) launchHostScans() chan scanReq {
 	s.ap.wg.Add(1)
 	go func() {
 		defer s.ap.wg.Done()
+		defer close(reqChan)
 
 		var offset int
-		var exhausted bool
+		var launched int
 		cutoff := time.Now().Add(-s.scanMinInterval)
-		for !s.ap.isStopped() && !exhausted {
+		for !s.ap.isStopped() {
```

Comment:
Do we need this loop? Isn't a single call to `HostsForScanning` enough to fetch a full batch?